docs/architecture/membership-subscriptions.md

```
@@ -0,0 +1,410 @@
+ACTIVE/PAUSED/PAST_DUE ──(cancel_subscription, immediate=False)──► (unchanged, cancel_at_period_end=True)
+
+EXPIRED ──(revive_subscription within window)──► ACTIVE (OFFLINE) / PENDING (ONLINE, awaits invoice.paid)
+```
+
+Key invariants:
+
+- Terminal subscriptions (`CANCELLED`, `EXPIRED`) **never** advance their
+  period. `record_payment` raises 400; `_apply_stripe_price_swap`
+  short-circuits.
+- Subscription tier wins over member tier on the `OrganizationMember` sync
+  signal — see [Member Sync](#member-sync).
```

markdownlint-cli2 (0.22.1)

[warning] Link fragments should be valid

(MD051, link-fragments)